tests/queries/0_stateless/01881_join_on_conditions_merge.sql.j2

```diff
@@ -58,20 +58,20 @@ SELECT DISTINCT t1.id FROM t1 INNER ANY JOIN t2 ON t1.id == t2.id AND toNullable
 -- `e + 1` is UInt16
-SELECT DISTINCT t1.key, toUInt8(t1.id) as e FROM t1 INNER ANY JOIN t2 ON t1.id == t2.id AND e + 1; -- { serverError INVALID_JOIN_ON_EXPRESSION }
+SELECT DISTINCT t1.key, toUInt8(t1.id) as e FROM t1 INNER ANY JOIN t2 ON t1.id == t2.id AND e + 1; -- { serverError INVALID_JOIN_ON_EXPRESSION,ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER }
 SELECT DISTINCT t1.key, toUInt8(t1.id) as e FROM t1 INNER ANY JOIN t2 ON t1.id == t2.id AND toUInt8(e + 1);
 
 SELECT '--';
 SELECT t1.id, t1.key, t1.key2, t2.id, t2.key, t2.key2  FROM t1 FULL JOIN t2 ON t1.id == t2.id AND t2.key == t2.key2 AND t1.key == t1.key2 ORDER BY t1.id NULLS FIRST, t2.id NULLS FIRST;
 
-SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t1.id; -- { serverError INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t2.id; -- { serverError INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t1.id + 2; -- { serverError INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t2.id + 2; -- { serverError INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t1.key; -- { serverError ILLEGAL_TYPE_OF_ARGUMENT, INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t2.key; -- { serverError ILLEGAL_TYPE_OF_ARGUMENT, INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 JOIN t2_nullable as t2 ON t2.key == t2.key2 AND (t1.id == t2.id OR isNull(t2.key2)); -- { serverError NOT_IMPLEMENTED,INVALID_JOIN_ON_EXPRESSION }
-SELECT * FROM t1 JOIN t2 ON t2.key == t2.key2 OR t1.id == t2.id; -- { serverError INVALID_JOIN_ON_EXPRESSION }
+SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t1.id; -- { serverError INVALID_JOIN_ON_EXPRESSION,ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER }
+SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t2.id; -- { serverError INVALID_JOIN_ON_EXPRESSION,ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER }
+SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t1.id + 2; -- { serverError INVALID_JOIN_ON_EXPRESSION,ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER }
+SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t2.id + 2; -- { serverError INVALID_JOIN_ON_EXPRESSION,ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER }
+SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t1.key; -- { serverError ILLEGAL_TYPE_OF_ARGUMENT, INVALID_JOIN_ON_EXPRESSION,ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER }
+SELECT * FROM t1 INNER ALL JOIN t2 ON t1.id == t2.id AND t2.key; -- { serverError ILLEGAL_TYPE_OF_ARGUMENT, INVALID_JOIN_ON_EXPRESSION,ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER }
+SELECT * FROM t1 FULL JOIN t2_nullable as t2 ON t2.key == t2.key2 AND (t1.id == t2.id OR isNull(t2.key2)); -- { serverError NOT_IMPLEMENTED,INVALID_JOIN_ON_EXPRESSION,ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER }
+SELECT * FROM t1 JOIN t2 ON t2.key == t2.key2 OR t1.id == t2.id; -- { serverError NOT_IMPLEMENTED,INVALID_JOIN_ON_EXPRESSION,ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER }
```

Comment:
Even if I could understand why there are two type of errors for other queries I could not understand why this query has 3 possible errors. Why?

Reply:
It should be two at maximum for old and new analyzer, we run both in CI, otherwise should be stable, I updated test accordingly.